src/lib/AnalysisApps/GWAS/Utils/cohortApi.test.tsx

```diff
@@ -0,0 +1,307 @@
+  useGetSourceIdQuery,
+  useGetCovariatesQuery,
+  useGetCovariateStatsQuery,
+  useGetConceptStatsByHareSubsetQuery,
+  useGetHistogramInfoQuery,
+  useGetSimpleOverlapInfoQuery,
+} from './cohortApi';
+
+const server = setupServer();
+
+const sourcesData = {
+  sources: [{ source_id: 123, source_name: 'MVP-batch19000101' }],
+};
+
+const cohortDefinitionAndStatsData = {
+  cohort_definitions_and_stats: [
+    {
+      cohort_definition_id: 573,
+      cohort_name: 'team2 - test new cohort - catch all',
+      size: 70240,
+    },
+    {
+      cohort_definition_id: 559,
+      cohort_name: 'test new cohort - catch all',
+      size: 70240,
+    },
+    {
+      cohort_definition_id: 574,
+      cohort_name: 'team2 - test new cohort - medium + large',
+      size: 23800,
+    },
+    {
+      cohort_definition_id: 575,
+      cohort_name: 'team2 - test new cohort - small',
+      size: 80,
+    },
+  ],
+};
+
+const covariatesData = { covariates: [{ concept_id: 123, concept_name: 'Test Covariate' }] };
+const covariateStatsData = { some_stat: 42 };
+const conceptStatsData = { concept_stats: [{ size: 10, concept_name: 'Example' }] };
+const histogramData = { bins: [{ start: 0, end: 10, personCount: 5 }] };
+const overlapData = { cohort_overlap: { case_control_overlap: 15 } };
+
+describe('cohortApi', () => {
+  beforeAll(() => {
+    // Start the interception.
+    server.listen();
+  });
+  beforeEach(() => {});
+
+  afterEach(() => {
+    // Remove any handlers you may have added
+    // in individual tests (runtime handlers).
+    server.resetHandlers();
+  });
+
+  afterAll(() => {
+    // Disable request interception and clean up.
+    server.close();
+  });
+
+  it('fetches and returns cohort definitions successfully', async () => {
+    const sourceId = '1234567890';
+    const selectedTeamProject = 'project2';
+
+    server.use(
+      http.get(
+        `${GEN3_COHORT_MIDDLEWARE_API}/cohortdefinition-stats/by-source-id/1234567890/by-team-project`,
+        ({ request }) => {
+          const url = new URL(request.url);
+          const project = url.searchParams.get('team-project');
+          if (project !== 'project2') {
+            return new HttpResponse(null, {
+              status: 403,
+            });
+          }
+          return HttpResponse.json(cohortDefinitionAndStatsData);
+        },
+      ),
+    );
+
+    const { result } = renderHook(() =>
+      useGetCohortDefinitionsQuery({ sourceId, selectedTeamProject }),
+    );
+
+    expect(result.current.isFetching).toBe(true);
+
+    await waitFor(() => expect(result.current.isSuccess).toBeTruthy());
+
+    expect(result.current).toMatchObject({
+      isError: false,
+      isFetching: false,
+      isSuccess: true,
+      isLoading: false,
+      data: cohortDefinitionAndStatsData,
+    });
+  });
+
+  it('returns error if project id not accessible ', async () => {
+    const sourceId = '1234567890';
+    const selectedTeamProject = 'project2345';
+
```

Comment:
Optional - consider mocking the console.error so the console doesn't print an error when running the test. See the console error might be confusing to developers, as they might think something is wrong when everything is actually behaving as expected. 

I think adding this will prevent the error from being printed while running the test: 

```  jest.spyOn(console, 'error').mockImplementation(() => {});```